src/utils/scraper.ts

```diff
@@ -10,42 +10,58 @@ async function launchBrowser(): Promise<Browser> {
   return browser;
 }
 
+async function withTimeout<T>(promise: Promise<T>, timeout: number): Promise<T> {
+  return Promise.race([
+    promise,
+    new Promise<never>((_, reject) =>
+      setTimeout(() => reject(new Error('Operation timed out')), timeout)
+    ),
+  ]);
+}
+
 export async function getLinkWithPuppeteer(
   url: string,
   linkQuerySelector: string,
-  cookies: CookieParam[] = []
+  cookies: CookieParam[] = [],
+  timeout: number = 4000 // Default timeout of 4 seconds
 ) {
   const browser = await launchBrowser();
   const page: Page = await browser.newPage();
 
-  await page.setUserAgent(
-    'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:124.0) Gecko/20100101 Firefox/124.0'
-  );
+  try {
+    await page.setUserAgent(
+      'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:124.0) Gecko/20100101 Firefox/124.0'
+    );
```

Comment:
For scrapping, I would recommend to randomize User Agent to bypass some anti-scrapping rules (scrapper would fail less)

Reply:
it was like this in the past, after some problems with the library we were using, I opted to keep it simple. But this was just a temp fix. We will get rid of puppeteer, as it is only being used for youtube, and now the API seems to be more than okay fort the links fallback